### DISCUSSION TITLE
Code blocks appear as (underlined) links

### DISCUSSION THREAD

**Discussion by miraculixx** [community/community#113792] [Repositories] [answered]
### Select Topic Area

Product Feedback

### Body

For some reason code blocks in restructuredtext-formatted README.rst files started to appear as underlined links recently:

![image](https://github.com/community/community/assets/859611/a614a0da-60f5-4eeb-ab32-00e460f68d57)

This can be switched off by changing to "Hide link underlines" in settings > accessibility > content. However imho this is not accessibility question, but an issue with the generation of the respective code block's html, that is not every line should be rendered inside an `<a>` tag, as lines of codes are not links.

For reference, the respective code block is valid RST ([source](https://github.com/omegaml/minibatch/blob/master/README.rst?plain=1)). Other README.rst in other repositories have the same issue, e.g. [Sphinx](https://github.com/sphinx-doc/sphinx/blob/d91ba115bdd47c265e082c35a7d39d5455687431/README.rst#L51), 

```
.. code:: python

      $ pip install minibatch
      $ docker run -d -p 27017:27017 mongo
```

### Related Issues

It appears that this specific issue (code block formatting) has been solved by github. Please report any other RST rendering issues at [Possible regression: failed rendering of ReStructuredText files #86715](https://github.com/orgs/community/discussions/86715#discussioncomment-9149558)

**Comment by vickkie** [comment]
### Hello @miraculixx 

- instead of using .. code:: python 
- try to use quotations like  ``````
- 3 infront --message inside -- 3 after

This way 
1. Install and setup

   ```
      pip install minibatch
      docker run -d -p 27017:27017 mongo
  ```
-   See extras & optional dependencies below to select specific packages according
   to your deployment needs, e.g. for MQTT, Kafka, omega|ml

**Reply by smheidrich** [reply]
That's for Markdown files. This is about reStructuredText (RST) files, where ``` is meaningless.

**Comment by vickkie** [comment]
ooh okay i get it

- You could  try using the .. code:: directive without specifying a language to see if it will workaround the issue

```
 .. code::

    $ pip install minibatch
    $ docker run -d -p 27017:27017 mongo
```

**Reply by miraculixx** [reply]
Thanks. However my point is it used to work as in my OP,  and it still should as afaik it is the official RST syntax for code blocks.

**Reply by vickkie** [reply]
Yeah the RST syntax  .. code:: language is collect. I think the issue is the current GitHub RST parser related,
It used to  work , lets see if issue will change in the future.  For now the specification on GitHub doesn't work

**Comment by smheidrich** [comment]
Possibly related: GitHub seems to have issues with RST more generally (e.g. files not rendering at all): https://github.com/orgs/community/discussions/86715

**Reply by felixfontein** [reply]
Indeed, and the problem with the underlines affects a lot of other documents as well. I'm for now happy that the documents which simply stopped rendering at all since beginning of January are rendering again, but having them rendered with proper syntax highlighting instead of underlines would be even better :)

**Comment by fangq** [comment]
two of my projects also had rendering issues for the README.rst files

https://github.com/NeuroJSON/zmat/blob/master/README.rst
[the other one ](https://github.com/fangq/jsonlab/blob/36fef2c804463c8357f343c3a70190e9af6d6ce1/README.rst)

**Reply by DimitriPapadopoulos** [reply]
Same here, see:
https://github.com/codespell-project/codespell/pull/3390#issuecomment-2020637386

**Reply by lukasturcani** [reply]
same!  https://github.com/lukasturcani/atomlite

**Reply by bartandrews** [reply]
Same here: https://github.com/HofstadterTools/HofstadterTools?tab=readme-ov-file#how-to-cite

**Reply by bartandrews** [reply]
Also, github has at the same time broken the alignment of my images in the readme.rst: https://github.com/HofstadterTools/HofstadterTools?tab=readme-ov-file#quick-start They should each be 32% width, as correctly rendered by sphinx: https://hofstadter.tools/overview.html#quick-start

**Reply by bartandrews** [reply]
my other code blocks, which are not underlined, have lost their syntax highlighting

**Reply by BrianPugh** [reply]
same https://github.com/BrianPugh/autoregistry/blob/main/README.rst

**Reply by miraculixx** [reply]
same https://github.com/dfunckt/django-rules?tab=readme-ov-file#using-rules 

(no affiliation)

**Comment by 12rambau** [comment]
I'll add mine to the piles, all my projects using a README.rst are facing issues: 
- every code-block get's underlined
- svg badges get a non existing line-break
- image directives are not using the display otpions

few problematic repositories: 
- https://github.com/readthedocs/sphinx-autoapi
- https://github.com/12rambau/sepal_ui
- https://github.com/gee-community/ipygee
- https://github.com/12rambau/pypackage

Strangly enough one of my repo (using the same readme template) is rendering perfectly now: https://github.com/gee-community/geetools
The only difference with the pthers is that I made a push very recently. Could it be just a caching issue on Github side ?

**Reply by miraculixx** [reply]
Thanks for this. It appears that some issues have been [fixed](https://github.com/orgs/community/discussions/113792#discussioncomment-9166369). Please report remaining issues here https://github.com/orgs/community/discussions/86715

**Comment by bckohan** [comment]
Just adding my upvote for a fix. I have 6 projects that are affected.

**Comment by casperdcl** [comment]
I opened a ticket (No. 2709825) with GitHub support.

https://github.com/tqdm/tqdm#readme looks sad now :(

Also related to https://github.com/tqdm/tqdm/issues/1546.

**Comment by pbelskiy** [comment]
Same problem for a long time ...

**Comment by enzbus** [comment]
Not sure if relevant, but on one of my projects it just happened today after I made (unrelated) edits to the `README.rst`, it was rendering correctly until before the edit. On another project of mine it was broken already a few weeks ago :(

**Comment by Gatsby-Lee** [comment]
so I am not alone in this issue.
I thought I did something wrong

**Comment by MauricioSalazare** [comment]
I second this request. RST readme texts are broken in my projects.

**Comment by 5j9** [comment, marked as answer]
Dear fellow Githubians, it seems that the rst Gods have answered our prayers! Today, I can see syntax highlights on my repositories once again!

**Reply by bartandrews** [reply]
Fixed for me too! :+1: However, the image formatting is still broken: https://github.com/HofstadterTools/HofstadterTools?tab=readme-ov-file#quick-start I used to have 3 images side-by-side, as correctly rendered by Sphinx: https://hofstadter.tools/overview.html#quick-start

**Reply by 12rambau** [reply]
the rest of the issues raised in this thread (images, code block and links) are still very wrong in my repositories as well.

**Reply by miraculixx** [reply]
I think it would be best to report all remaining issues here https://github.com/orgs/community/discussions/86715

**Comment by pbelskiy** [comment]
It's still problem with tables -- https://github.com/pbelskiy/ujenkins
<img width="1098" alt="Screenshot 2024-05-05 at 14 18 06" src="https://github.com/community/community/assets/8102549/157627a8-e9b0-4fdc-8c40-4246020aba41">
